src/main/java/org/minions/devfund/reinaldo/battleship/Ocean.java

```diff
@@ -0,0 +1,183 @@
+                int column = random.nextInt(SHIPLENGTH);
+                boolean horizontal = random.nextBoolean();
+                placeShip = factory.getShip(newShip).okToPlaceShipAt(row, column, horizontal, this);
+                if (placeShip) {
+                    factory.getShip(newShip).placeShipAt(row, column, horizontal, this);
+                }
+            }
+            while (!placeShip);
+        });
+
+    }
+
+    /**
+     * Looks the position if it is exist on that position.
+     *
+     * @param row    int row position.
+     * @param column int column position.
+     * @return true if position was occupied.
+     */
+    boolean isOccupied(final int row, final int column) {
+        return ships[row][column].getLength() > 1;
+    }
+
+    /**
+     * Shoots a position.
+     *
+     * @param row    int row position.
+     * @param column int column position.
+     * @return true if ship was shot.
+     */
+    boolean shootAt(final int row, final int column) {
+        shotsFired++;
+        if (ships[row][column].shootAt(row, column)) {
+            increaseHitCounters(row, column);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Uptates the hit counters and sunk.
+     *
+     * @param row    row position.
+     * @param column column position.
+     */
+    private void increaseHitCounters(final int row, final int column) {
+        if (ships[row][column].isSunk()) {
+            shipsSunk++;
+        }
+        hitCount++;
+    }
+
+    /**
+     * Gets the shots fired.
+     *
+     * @return int shots fired quantity.
+     */
+    int getShotsFired() {
+        return shotsFired;
+    }
+
+    /**
+     * Gets the hit count.
+     *
+     * @return int he hit number.
+     */
+    int getHitCount() {
+        return hitCount;
+    }
+
+    /**
+     * Gets the ships sunk quantity.
+     *
+     * @return int ships sunk number.
+     */
+    int getShipsSunk() {
+        return shipsSunk;
+    }
+
+    /**
+     * if the game over to battleship.
+     *
+     * @return true if game over.
+     */
+    boolean isGameOver() {
+        return shipsSunk == SHIPS_QUANTITY;
+    }
+
+    /**
+     * Gets the ship array.
+     *
+     * @return matrix of ship
+     */
+    Ship[][] getShipArray() {
+        return ships;
+    }
+
+    /**
+     * Verifies if the position at the border on ocean.
+     *
+     * @param row    int row position.
+     * @param column int column position.
+     * @return true if border is occupied.
+     */
+    boolean isBorderOccupied(final int row, final int column) {
+        final int startRow = row + 1 < SHIPLENGTH ? row + 1 : row;
+        final int endRow = row - 1 >= 0 ? row - 1 : row;
+        final int startColumn = column + 1 < SHIPLENGTH ? column + 1 : column;
+        final int endColumn = column - 1 >= 0 ? column - 1 : column;
+        return isOccupied(startRow, column) || isOccupied(startRow, startColumn) || isOccupied(startRow, endColumn)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (7) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1067)